static/js/compose_actions.js

```diff
@@ -365,6 +365,14 @@ exports.on_topic_narrow = function () {
     $('#compose-textarea').focus().select();
 };
 
+exports.quote_and_reply = function (opts) {
+    exports.respond_to_message(opts);
+    var message = current_msg_list.selected_message();
+    var quoted_message = '```quote\n' + message.content + '\n```\n';
+    $('#compose-textarea').val(quoted_message);
+    $("#compose-textarea").trigger("autosize.resize");
+};
```

Comment:
This function is a duplicate of similar code in `popovers.js`.  We avoid code duplication in the Zulip project, since it very often leads to bugs.  Please refactor so that the `popovers.js` case can call the same function that this does (perhaps with an argument passed in).

Finally, `message.content` is the HTML; we want the raw markdown that the original user entered; see the similar code in `popovers` for how to fetch the raw content from the server.

Reply:
@timabbott Updated.